### PR TITLE
nestjs backend

### PR DESCRIPTION
**Pull request by GuanglongDu**

### CODE REVIEW

File: src/user/user.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { UserService } from './user.service';
+import { MongooseModule } from '@nestjs/mongoose';
+import { UserSchema } from '../schema/user/user.schema';
+import { Logger } from '@nestjs/common';
```
Comment (rankun203) [inline src/user/user.service.spec.ts]: Unused variable.

File: README.md [hunk 2]
```diff
@@ -71,3 +71,34 @@ Fork this repo and send us a pull request when you think it's ready for review.
...
+$ npm run start:prod
+```
+
+## Running test case
+```bash
+$ npm run test
+```
+## Swagger UI
+```
+http://localhost:3000/api
```
Comment (rankun203) [inline README.md]: I have another project running on 3000, after I launched this server how do I access it? I don't see logs in console.

File: src/follows/follows.service.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        if (user.id === followId) {
+            throw new UnauthorizedException('Follow yourself');
+        }
+        const follow = await this.userService.findById(followId);
+        if (!follow) {
+            throw new UnauthorizedException('User does not exists');
+        }
+        const exists = await this.findOne({ uid: Types.ObjectId(user.id), followId: Types.ObjectId(followId) });
+        if (exists) {
+            throw new UnauthorizedException('User had followed');
```
Comment (rankun203) [inline src/follows/follows.service.ts]: Was this [UnauthorizedException](https://github.com/nestjs/nest/blob/master/packages/common/exceptions/unauthorized.exception.ts) used by a mistake?

File: src/user/user.service.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        return this.userToken(user);
+    }
+
+    private userToken(user: User): string {
+        const payload: JWTPayload = {
+            uid: user.id,
+            password: user.password,
+        };
+
+        const token = jwt.sign(payload, 'test-backend', {
```
Comment (rankun203) [inline src/user/user.service.ts]: Same here, It's better to separate configurations from code, usually we use environment variables to let the deployment environment to specify it.

File: src/app.module.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import { Module } from '@nestjs/common';
+import { MongooseModule } from '@nestjs/mongoose';
+import { FollowsModule } from './follows/follows.module';
+import { UserModule } from './user/user.module';
+import { PassportModule } from './common/passport/passport.module';
+@Module({
+  imports: [
+    MongooseModule.forRoot('mongodb://localhost:27017/test-backend', { useNewUrlParser: true, useUnifiedTopology: true, useCreateIndex: true }),
```
[the remark this answers is not in the document]
Comment (GuanglongDu) [reply src/app.module.ts]: Should be add ConfigModule to load env file, eg: dev.env,testing.env

File: src/user/user.service.ts (hunk 4 above)
Comment (GuanglongDu) [reply src/user/user.service.ts]: yes, i didn't config config module

File: src/user/user.service.spec.ts (hunk 1 above)
Comment (GuanglongDu) [reply src/user/user.service.spec.ts]: forgot remove it.

File: src/follows/follows.service.ts (hunk 3 above)
Comment (GuanglongDu) [reply src/follows/follows.service.ts]: yes,there just record error message.  need created new exception with error message and error code extends http exception.

Comment (rankun203) [reply src/follows/follows.service.ts]: What do you mean by error code? Can you make an example?

Comment (GuanglongDu) [reply src/follows/follows.service.ts]: The error defined as follows:
export class UserIsNotExistsError extends HttpException {
    constructor() {
        super('user not exists', ErrorCode.USER_IS_NOT_EXISTS|| 10010);
    }
}

File: README.md (hunk 2 above)
Comment (GuanglongDu) [reply README.md]: the application start success when the console log "Nest application successfully started "。

File: src/app.module.ts [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+    PassportModule,
+    UserModule,
+    FollowsModule],
+})
```
[the remark this answers is not in the document]
Comment (GuanglongDu) [reply src/app.module.ts]: I don't know, save and format file again, still not changed.

[the remark this answers is not in the document]
Comment (rankun203) [reply src/app.module.ts]: Do a `npm run format` and you will see.

File: src/follows/follows.service.ts (hunk 3 above)
Comment (rankun203) [reply src/follows/follows.service.ts]: OK,

BTW, you can use 3 back ticks to define code blocks.

File: README.md (hunk 2 above)
Comment (rankun203) [reply README.md]: OK

**Comment by xavierchow** [comment]
Thanks all, the PR is good enough for our evaluation, will get back to you soon.
closing.

**Pull request closed without merging**